### PR TITLE
Modify qchem openfermion Hamiltonian complex coefficient tolerance

### PR DESCRIPTION
**Pull request by soranjh**
**Context:**
The `import_operator` function, which converts an openfermion operator to a PennyLane operator, has a tolerance for discarding the imaginary part of the coefficients of the input operator. Coefficients with imaginary part less than the tolerance are considered to be real. The default tolerance is currently  `2.22e-08` which makes the conversion to fail often. This PR decreases the tolerance to `2.22e-06` and converts the error to a warning when the input openfermion operator has a coefficient with an imaginary component smaller than this value.

**Description of the Change:**

- The default value for the `tol` argument is changed from `1e08` to `1e010`. This value will be used in `numpy.real_if_close` as `tol * machine-epsilon`.
- The error is changed to warning.

**Benefits:**
The user is informed about having non-real coefficients in the operator without failing the conversion. 

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: pennylane/qchem/convert.py [hunk 1]
```diff
@@ -141,7 +143,7 @@ def _openfermion_to_pennylane(qubit_operator, wires=None):
     (array([0.1, 0.2]), [PauliX(wires=['w0']), PauliY(wires=['w0']) @ PauliZ(wires=['w2'])])
     """
     n_wires = (
-        1 + max([max([i for i, _ in t]) if t else 1 for t in qubit_operator.terms])
```
Comment (soranjh) [inline pennylane/qchem/convert.py:144]: Fix requested by pylint.

File: pennylane/qchem/openfermion_obs.py [hunk 2]
```diff
@@ -953,7 +957,7 @@ def molecular_hamiltonian(
 
     h_of, qubits = (decompose(hf_file, mapping, core, active), 2 * len(active))
 
-    h_pl = qml.qchem.convert.import_operator(h_of, wires=wires)
+    h_pl = qml.qchem.convert.import_operator(h_of, wires=wires, tol=convert_tol)
```
Comment (obliviateandsurrender) [inline pennylane/qchem/openfermion_obs.py:960]: 🚀🚀

File: pennylane/qchem/openfermion_obs.py [hunk 3]
```diff
@@ -870,6 +871,9 @@ def molecular_hamiltonian(
         grouping_type (str): method to group commuting observables
         grouping_method (str): the graph coloring heuristic to use in solving minimum clique cover
             for grouping
+        convert_tol (float): Tolerance in machine epsilons for the imaginary part of the
```
Comment (obliviateandsurrender) [inline pennylane/qchem/openfermion_obs.py:874]: We might want to provide a link to the machine epsilons here? (if it exists)

**Review by obliviateandsurrender — CHANGES_REQUESTED** [review]
Looking good @soranjh, leaving a couple of suggestions.

**Review by obliviateandsurrender — APPROVED** [review]

**Pull request merged**